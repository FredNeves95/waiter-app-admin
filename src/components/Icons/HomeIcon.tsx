import { SvgIconProps } from '../../utils/types/icons';

export const HomeIcon = ({ className, color }: SvgIconProps) => (
  <svg
    className={className}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 24 24"
    fill="none"
  >
    <path fillRule="evenodd" clipRule="evenodd" d="M23.0377 8.64827L15.3577 2.67451C13.9715 1.59547 12.0297 1.59547 10.6422 2.67451L2.96222 8.64827C2.02654 9.37659 1.47998 10.4953 1.47998 11.6793V21.0656C1.47998 23.1865 3.19902 24.9056 5.31998 24.9056H20.68C22.8009 24.9056 24.52 23.1865 24.52 21.0656V11.6793C24.52 10.4953 23.9734 9.37659 23.0377 8.64827Z" stroke={color || 'currentColor'} strokeWidth="2"/>
  </svg>
);
