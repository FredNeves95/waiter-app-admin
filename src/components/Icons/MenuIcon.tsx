import { SvgIconProps } from '../../utils/types/icons';

export const MenuIcon = ({ className, color }: SvgIconProps) => (
  <svg
    className={className}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 24 24"
    fill="none"
  >
    <path d="M13 12.5H8M12 15.5H9M4 1.5V19.5M3 19.5H15C16.1046 19.5 17 18.6046 17 17.5V3.5C17 2.39543 16.1046 1.5 15 1.5H3C1.89543 1.5 1 2.39543 1 3.5V17.5C1 18.6046 1.89543 19.5 3 19.5ZM12.5 7.5C12.5 8.60457 11.6046 9.5 10.5 9.5C9.39543 9.5 8.5 8.60457 8.5 7.5C8.5 6.39543 9.39543 5.5 10.5 5.5C11.6046 5.5 12.5 6.39543 12.5 7.5Z" stroke={color || 'currentColor'} strokeWidth="1.5" strokeLinecap="round" strokeLinejoin="round"/>
  </svg>
);
