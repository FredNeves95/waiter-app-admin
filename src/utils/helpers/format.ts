export const formatToBRCurrency = new Intl.NumberFormat('pt-BR', {
  style: 'currency',
  currency: 'BRL'
});
