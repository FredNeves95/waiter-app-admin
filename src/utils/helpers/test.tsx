import 'jsdom-global/register';
import { mount, shallow,   } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { ReactNode } from 'react';
import { ActivePageProvider } from '../../contexts/ActivePageContext';

export function mountWithTheme(children: ReactNode) {
  return mount(
    <ThemeProvider theme={theme}>
      <ActivePageProvider>
        {children}
      </ActivePageProvider>
    </ThemeProvider>);
}

export function shallowWithTheme(children: ReactNode) {
  return shallow(
    <ThemeProvider theme={theme}>
      <ActivePageProvider>
        {children}
      </ActivePageProvider>
    </ThemeProvider>);
}
