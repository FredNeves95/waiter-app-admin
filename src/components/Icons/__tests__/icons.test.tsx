import { describe, expect, it } from 'vitest';
import { mountWithTheme } from '../../../utils/helpers/test';
import { HistoryIcon } from '../HistoryIcon';
import { HomeIcon } from '../HomeIcon';
import { LogoutIcon } from '../LogoutIcon';
import { MenuIcon } from '../MenuIcon';
import { ProfileIcon } from '../ProfileIcon';
import { RestartIcon } from '../Restarticon';
import { UsersIcon } from '../UsersIcon';

const renderIcons = () => mountWithTheme(<>
  <HistoryIcon/>
  <HomeIcon/>
  <LogoutIcon/>
  <MenuIcon/>
  <ProfileIcon/>
  <RestartIcon/>
  <UsersIcon/>
</>);

describe('Icons tests', () => {
  it('Should look like snapshots', () => {
    const icons = renderIcons();

    expect(icons.html()).toMatchSnapshot();
  });
});
