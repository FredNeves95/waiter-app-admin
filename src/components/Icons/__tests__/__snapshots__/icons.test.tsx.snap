// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icons tests > Should look like snapshots 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\"><path d=\\"M13 7.5H5M13 10.4999H5M10 13.5H5M16.449 18.896L15.459 19.3935C15.1785 19.5345 14.8481 19.5355 14.5667 19.3964L12.9942 18.6188L11.4533 19.3929C11.172 19.5342 10.8406 19.5348 10.5588 19.3945L9 18.6185L7.44124 19.3945C7.15942 19.5348 6.82799 19.5342 6.54668 19.3929L5.00578 18.6188L3.4333 19.3964C3.15191 19.5355 2.82151 19.5345 2.54102 19.3935L1.55102 18.896C1.21321 18.7263 1.00001 18.3805 1 18.0025V2.99752C1 2.61947 1.2132 2.27374 1.551 2.104L2.541 1.60648C2.82149 1.46552 3.15189 1.46446 3.43328 1.60361L5.00576 2.3812L6.54666 1.60713C6.82797 1.46581 7.1594 1.46521 7.44122 1.60551L9 2.38151L10.5588 1.60551C10.8406 1.46521 11.172 1.46581 11.4533 1.60713L12.9942 2.3812L14.5667 1.60361C14.8481 1.46446 15.1785 1.46552 15.459 1.60648L16.449 2.104C16.7868 2.27374 17 2.61946 17 2.99752V18.0025C17 18.3806 16.7868 18.7263 16.449 18.896Z\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path></svg><svg width=\\"20\\" height=\\"20\\" viewBox=\\"0 0 20 20\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M17.5283 6.48621L11.7683 2.00589C10.7286 1.19661 9.27231 1.19661 8.23167 2.00589L2.47167 6.48621C1.76991 7.03245 1.35999 7.87149 1.35999 8.75949V15.7992C1.35999 17.3899 2.64927 18.6792 4.23999 18.6792H15.76C17.3507 18.6792 18.64 17.3899 18.64 15.7992V8.75949C18.64 7.87149 18.2301 7.03245 17.5283 6.48621Z\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\"><path d=\\"M14.657 3.34302C17.781 6.46702 17.781 11.533 14.657 14.657C11.533 17.781 6.467 17.781 3.343 14.657C0.219 11.533 0.219 6.46702 3.343 3.34302M9 1V9\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\"><path d=\\"M13 12.5H8M12 15.5H9M4 1.5V19.5M3 19.5H15C16.1046 19.5 17 18.6046 17 17.5V3.5C17 2.39543 16.1046 1.5 15 1.5H3C1.89543 1.5 1 2.39543 1 3.5V17.5C1 18.6046 1.89543 19.5 3 19.5ZM12.5 7.5C12.5 8.60457 11.6046 9.5 10.5 9.5C9.39543 9.5 8.5 8.60457 8.5 7.5C8.5 6.39543 9.39543 5.5 10.5 5.5C11.6046 5.5 12.5 6.39543 12.5 7.5Z\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\"><path d=\\"M12.1 6.40002C13.3 7.60002 13.3 9.50002 12.1 10.6C10.9 11.7 9.00002 11.8 7.90002 10.6C6.80002 9.40002 6.70002 7.50002 7.90002 6.40002C9.10002 5.30002 10.9 5.20002 12.1 6.40002M3.90002 17C5.50002 15.5 7.60002 14.5 10 14.5C12.3 14.5 14.5 15.4 16.1 17M2 14C1.4 12.8 1 11.4 1 10C1 5 5 1 10 1C15 1 19 5 19 10C19 11.4 18.6 12.8 18 14\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path></svg><svg width=\\"26\\" height=\\"26\\" viewBox=\\"0 0 26 26\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M13 21.0033C15.9314 21.0035 18.6284 19.4012 20.0299 16.8265\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path><path d=\\"M21.0033 13C21.0033 8.57992 17.4201 4.9967 13 4.9967\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path><path d=\\"M13 4.9967C10.0686 4.99646 7.37166 6.59883 5.97009 9.17344\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path><path d=\\"M4.99664 13C4.99664 17.4201 8.57986 21.0033 13 21.0033\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path><path d=\\"M5.97009 9.17335C7.71956 5.95833 11.4272 4.33982 14.9743 5.2427C18.5213 6.14558 21.0039 9.33975 21.0034 12.9999\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path><path d=\\"M4.99664 13C4.99608 16.6602 7.47866 19.8544 11.0257 20.7572C14.5728 21.6601 18.2804 20.0416 20.0299 16.8266\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path><path d=\\"M9.04532 9.17346H5.50885V5.63599\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path><path d=\\"M16.9547 16.8265H20.4911V20.364\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\"><circle cx=\\"8.99835\\" cy=\\"9.00854\\" r=\\"3.49145\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></circle><circle cx=\\"18.0021\\" cy=\\"9.99897\\" r=\\"2.50104\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></circle><path d=\\"M2.99585 20.5033V19.4859C2.99585 17.285 4.77959 15.5012 6.98051 15.5012H11.0162C13.2171 15.5012 15.0008 17.285 15.0008 19.4859V20.5033\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path><path d=\\"M18.0021 15.5012H19.1045C21.3055 15.5012 23.0892 17.285 23.0892 19.4859V20.5033\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path></svg>"`;
